cms/datasets/models.py

```diff
@@ -15,16 +17,66 @@
 
 # TODO: This needs a revisit. Consider caching + fewer requests
 class ONSDatasetApiQuerySet(APIQuerySet):
+    def __init__(self, *args: object, **kwargs: object) -> None:
+        super().__init__(*args, **kwargs)
+        self.token: str | None = None
+
+    def with_token(self, token: str) -> "ONSDatasetApiQuerySet":
+        """Return a cloned queryset with the given authentication token.
+
+        We clone the queryset to ensure the method is stateless and doesn't mutate
+        the original queryset. This allows chaining operations (e.g.,
+        ONSDataset.objects.with_token(token).filter(...).all()) without affecting
+        other code that may be using the same base queryset instance.
+
+        Args:
+            token: Bearer token for API authentication
+
+        Returns:
+            Cloned queryset with token attached
+        """
+        clone: ONSDatasetApiQuerySet = self.clone()
+        clone.token = token
+        return clone
+
     def get_results_from_response(self, response: Mapping) -> Iterable:
         results: Iterable = response["items"]
         return results
 
     def fetch_api_response(self, url: str | None = None, params: Mapping | None = None) -> dict:
-        api_response: dict = super().fetch_api_response(url=url, params=params)
+        # Add Authorization header if token is set
+        headers = dict(self.http_headers) if self.http_headers else {}
+        if self.token:
+            headers["Authorization"] = f"Bearer {self.token}"
```

Comment:
The value of the access token (from the cookie) you pass in will already include `Bearer` prefix, so this should simply be:
```py
headers["Authorization"] = self.token
```

This is an existing pattern in their auth flow. Just like we have done for bundles api client: https://github.com/ONSdigital/dis-wagtail/blob/main/cms/bundles/clients/api.py#L44
I suggest adding a comment in both places.